### PR TITLE
opensuse: populate the cache dir even when creating a sysext image

### PR DESCRIPTION
**Pull request by fbuihuu**

### CODE REVIEW

File: mkosi/distributions/opensuse.py [hunk 1]
```diff
@@ -94,6 +111,42 @@ def zypper_modifyrepo(state: MkosiState, repo: str, caching: bool) -> None:
...
+    # from the image and replaced with the final repositories at the end of the
+    # installation process.
+    #
+    # We need to enable packages caching in any cases to make sure that the
+    # package cache stays populated after "zypper install".
+
+    if state.config.local_mirror:
+        zypper_addrepo(state, state.config.local_mirror, "local-mirror", caching=True)
+    else:
+        for (name, url) in OpensuseInstaller.repositories(state):
```
Comment (behrmann) [inline mkosi/distributions/opensuse.py:126]: Style nit:
```suggestion
        for name, url in OpensuseInstaller.repositories(state):
```
same below

File: mkosi/distributions/opensuse.py [hunk 2]
```diff
@@ -94,6 +111,42 @@ def zypper_modifyrepo(state: MkosiState, repo: str, caching: bool) -> None:
...
+    # populate mkosi package cache.
+    for (name, url) in OpensuseInstaller.repositories(state):
+        if state.config.local_mirror and state.config.base_image is None:
+            zypper_addrepo(state, url, name)
+
+        elif state.config.base_image is None:
+            zypper_modifyrepo(state, name, caching=False)
+
+        else:
+            zypper_removerepo(state, name)
```
Comment (behrmann) [inline mkosi/distributions/opensuse.py:147]:
```suggestion
        if state.config.local_mirror and state.config.base_image is None:
            zypper_addrepo(state, url, name)
        elif state.config.base_image is None:
            zypper_modifyrepo(state, name, caching=False)
        else:
            zypper_removerepo(state, name)
```
no need for the extra vertical whitespace

**Review by behrmann — COMMENTED** [review]
Looks reasonable, two style nits.

**Comment by fbuihuu** [comment]
@behrmann, thanks for the quick review. I pushed a new version.

**Review by behrmann — APPROVED** [review]

**Comment by behrmann** [comment]
Nah, sorry, my bad, this has been open long enough, but I forgot about it.

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**